.github/workflows/ports_windows.yml

```diff
@@ -17,7 +17,130 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  build:
+  build-vs:
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: [x86, x64]
+        configuration: [Debug, Release]
+        variant: [dev, standard]
+        visualstudio: ['2017', '2019', '2022']
+        include:
+        - visualstudio: '2017'
+          runner: windows-latest
+          vs_version: '[15, 16)'
+        - visualstudio: '2019'
+          runner: windows-2019
+          vs_version: '[16, 17)'
+        - visualstudio: '2022'
+          runner: windows-2022
+          vs_version: '[17, 18)'
+        exclude:
+        - variant: standard
+          configuration: Debug
+    runs-on: ${{ matrix.runner }}
+    steps:
+    - name: Install Visual Studio 2017
```

Comment:
This is rather slow, could this be cached?

Reply:
I'm guessing that the installers poke windows registry, so probably not. Maybe there is a way to create a docker container with this pre-installed or something like that? 

(It's already still faster than the current appveyor build though)